backend/OR-Tools/OR-Tools/Solvers/OrToolsSolver.cs

```diff
@@ -0,0 +1,118 @@
+﻿using Domains.Models.Input;
+using Domains.Models.Output;
+
+using OR_Tools.Models;
+using OR_Tools.Interfaces;
+
+using Google.OrTools.ConstraintSolver;
+
+namespace OR_Tools.Solvers
+{
+    internal class OrToolsSolver : ISolver
+    {
+        private static readonly long MAX_VEHICLE_PENALTY      = 1000;
+        private static readonly long MAX_VEHICLE_ROUTING_TIME = 1000;
+
+        private readonly OrToolsConverter converter;
+        
+        public OrToolsSolver()
+        {
+            this.converter = new OrToolsConverter();
+        }
+
+        public FileOutput Solve(FileInput input)
+        {
+            return Solve(input, CreateSearchParameters());
+        }
+
+        public FileOutput Solve(FileInput input, RoutingSearchParameters searchParameters)
+        {
+            WorkData data = converter.ToWorkData(input);
+
+            RoutingIndexManager manager = new RoutingIndexManager(data.DistanceMatrix.GetLength(0), data.VehiclesAmount, data.Starts, data.Ends);
+            RoutingModel routing = new RoutingModel(manager);
+
+            AddCapacityConstrains(data, manager, routing);
+            AddTimeWindowConstrains(data, manager, routing);
+            AddPenaltiesAndDroppingVisits(data, manager, routing);
+
+            Assignment solution = routing.SolveWithParameters(searchParameters);
+
+            if (solution == null) return null;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/iamprovidence/CVRPWT/pullRequest?prid=4635982)